ledger/complete/wal/checkpoint_v6_writer.go

```diff
@@ -0,0 +1,529 @@
+package wal
+
+import (
+	"bufio"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/ledger/complete/mtrie/flattener"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/node"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+	utilsio "github.com/onflow/flow-go/utils/io"
+)
+
+const subtrieLevel = 4
+const subtrieCount = 1 << subtrieLevel
+
+type NodeEncoder func(node *trie.MTrie, index uint64, scratch []byte) []byte
+
+type resultStoringSubTrie struct {
```

Comment:
we might find a better naming for this one like `SubTrieStorable`

Reply:
It's actually the result of storing sub trie, not the subtrie itself to be stored. 

The result could be succeeded or failed, as you can see the result also contains `Err`.